services/dataAccessLayer.js

```diff
@@ -0,0 +1,34 @@
+const userQuery = require("../models/users");
+
+const retrieveUsers = async ({ id = null, usernames = null, req = null }) => {
+  if (id) {
+    const result = await userQuery.fetchUser({ userId: id });
+    removeSensitiveInfo(result.user);
+    return result;
+  } else if (usernames) {
+    const { users } = await userQuery.fetchUsers(usernames);
+    users.forEach((element) => {
+      removeSensitiveInfo(element);
+    });
+    return users;
+  } else {
+    const { allUsers, nextId, prevId } = await userQuery.fetchPaginatedUsers(req.query);
```

Comment:
You should check if req is null or not. 
Let's say someone calls your function ` retrieveUsers({})`, then `req.query` will throw an error here. 

Also since you are just using the query, then why not pass only the query instead of the entire request object?